controller/scheduler/scheduler.go

```diff
@@ -278,6 +288,34 @@ func (s *Scheduler) streamHostEvents() error {
 	}
 }
 
+func (s *Scheduler) streamRouterEvents() {
+	log := s.logger.New("fn", "streamRouterEvents")
+
+	var events chan *discoverd.Event
+	var stream stream.Stream
```

Comment:
The stream is closed in a deferred function call [here](https://github.com/flynn/flynn/blob/scheduler-graceful-shutdown/controller/scheduler/router.go#L78).